nntrainer/models/neuralnet.cpp

```diff
@@ -205,7 +205,10 @@ int NeuralNetwork::initialize() {
 
   initialized = true;
 
-  if (!load_path.empty()) {
+  // @note we need check loadedWeight for the case of multiple call of load to
+  // load weight. Only the weight needs to be loaded here. Becuase the buffer
+  // for the optimizer is not allocated yet.
+  if (!load_path.empty() && !loadedWeight) {
```

Comment:
is this possible? in the current design, the model cannot load weights before initialization as initialization allocates weight buffers.

Reply:
model_graph.allocateWeights() is called at line 204. so, the buffer for weight is allocated and so, it is possible to load weight.

Reply:
My concern is `loadedWeight` will always be false here, as the model cannot be initialized twice and NeuralNetwork::load() needs model to be initialized.